$OpenBSD: patch-src_FbTk_Timer_hh,v 1.1 2013/01/21 09:49:20 dcoppa Exp $

Simplifies and fix bugs in FbTk::Timer
(upstream git commit 33161595f482d0dad950ce127b6016eefe8ea691)

--- src/FbTk/Timer.hh.orig	Mon Dec 10 18:26:54 2012
+++ src/FbTk/Timer.hh	Mon Jan 21 10:25:31 2013
@@ -61,7 +61,7 @@ class Timer { (public)
 
     static void updateTimers(int file_descriptor);
 
-    int isTiming() const { return m_timing; }
+    int isTiming() const { return (m_start > 0); }
     int getInterval() const { return m_interval; }
 
     int doOnce() const { return m_once; }
@@ -77,7 +77,6 @@ class Timer { (public)
 private:
     RefCount<Slot<void> > m_handler; ///< what to do on a timeout
 
-    bool m_timing; ///< clock running?
     bool m_once;  ///< do timeout only once?
     int m_interval; ///< Is an interval-only timer (e.g. clock), in seconds
 
